### PR TITLE
DDR5 IO Power unit tests

### PR DESCRIPTION
**Pull request by alexcani**

### CODE REVIEW

File: src/DRAMPower/DRAMPower/standards/ddr5/interface_calculation_DDR5.h [hunk 1]
```diff
@@ -12,12 +12,11 @@ class DDR5;
 
 class InterfaceCalculation_DDR5 {
    public:
-    InterfaceCalculation_DDR5(DDR5 &ddr);
+    InterfaceCalculation_DDR5(const MemSpecDDR5 &memspec);
```
Comment (alexcani) [inline src/DRAMPower/DRAMPower/standards/ddr5/interface_calculation_DDR5.h:16]: This API change reinforces separation of concerns and removes the circular dependency between DDR5 and InterfaceCalculation_DDR5

File: src/DRAMPower/DRAMPower/memspec/MemSpecDDR5.cpp [hunk 2]
```diff
@@ -50,6 +51,9 @@ MemSpecDDR5::MemSpecDDR5(nlohmann::json &memspec)
     memPowerSpec[VPP].iXX2P     = (parseUdoubleWithDefault(memspec["mempowerspec"]["ipp2p"], "ipp2p"));
     memPowerSpec[VPP].iXX3P     = (parseUdoubleWithDefault(memspec["mempowerspec"]["ipp3p"], "ipp3p"));
 
+    // TODO: have different spec for vddq?
+    memPowerSpec.push_back(MemPowerSpec());
+    memPowerSpec[VDDQ] = memPowerSpec[VDD];
```
Comment (alexcani) [inline src/DRAMPower/DRAMPower/memspec/MemSpecDDR5.cpp:57]: @lsteiner-tukl what you think about this? We aren't parsing the VDDQ power domain, but we use it in the IO power calculation class. Right now I just made it the same as VDD but i'm not sure what's the correct approach

File: tests/tests_drampower/interface/test_interface_ddr5.cpp [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+    double t_CK;
+    double voltage;
+    std::unique_ptr<InterfaceCalculation_DDR5> io_calc;
+};
+
+TEST_F(DDR5_Energy_Tests, Parameters) {
+    ASSERT_TRUE(t_CK > 0.0);
+    ASSERT_TRUE(voltage > 0.0);
+}
+
```
Comment (alexcani) [inline tests/tests_drampower/interface/test_interface_ddr5.cpp:221]: @lsteiner-tukl please pay special attention to the 4 tests below, the goal is to validate the formula being used for the calculation, i.e. our understanding of the termination scheme

File: src/DRAMPower/DRAMPower/memspec/MemSpecDDR5.cpp (hunk 2 above)
Comment (lsteiner-tukl) [reply src/DRAMPower/DRAMPower/memspec/MemSpecDDR5.cpp:57]: We need to add it to the mempowerspec, the standard specified currents for all three voltage domains.

**Pull request merged**